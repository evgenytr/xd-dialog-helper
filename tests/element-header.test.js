const header = require('../src/main').types.HEADER;
const actions = {
    change: jest.fn(() => undefined),
    values: jest.fn(() => {}),
    close: jest.fn(() => {}),
    cancel: jest.fn(() => {}),
    registerElement: jest.fn(() => undefined)
};

const rendered = header.render('dialog', {
    id: 'header',
    label: 'Hello World',
    htmlAttributes: {
        someAttr: 'hello'
    }
}, actions);

describe('Header Element', () => {
    describe('render()', () => {
        it('should render correctly', () => {
            expect(rendered).toMatchSnapshot();
        });
    });
    describe('value()', () => {
        it('should correctly return undefined as value', () => {
            expect(header.value({wrapper: document.createElement('h2')})).toBeUndefined();
        });
    });
    describe('validate()', () => {
        it('should always return true', () => {
            expect(header.valid({})).toBeTruthy();
        });
    });
    describe('name', () => {
        expect(header.type).toMatchSnapshot();
    });
});
