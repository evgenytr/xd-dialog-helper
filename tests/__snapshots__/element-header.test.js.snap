// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Header"`;

exports[`Header Element render() should render correctly 1`] = `
Object {
  "input": undefined,
  "props": Object {
    "id": "hr",
    "label": "Hello World",
  },
  "type": Object {
    "render": [Function],
    "type": "Header",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <h2
    id="dialog-hr"
  >
    Hello World
  </h2>,
}
`;
